warehouse/keyword-common/src/main/java/datawave/util/keyword/KeywordExtractor.java

```diff
@@ -99,23 +102,37 @@ public void parseOptions(Map<String,String> iteratorOptions) {
         }
     }
 
+    @Nonnull
```

Comment:
Consider adding the javadoc:
```suggestion
    /**
     * Extract keywords from the <code>foundContent</code> field in priority order based on the list of views in <code>preferredViews</code> field. Will return
     * a KeywordResults object populated with the keywords extracted from the first view in <code>preferredViews</code> that is found to contain content. If no
     * keywords can be extracted from all preferred views, returns an empty <code>KeywordResults</code> object.
     *
     * @return a KeywordResults object containing keywords from the first of the <code>preferredViews</code> found in <code>foundContent</code> that yields a
     *         non-empty set of keywords or an empty KeywordResults object if no keywords can be extracted.
     */
    @Nonnull
```